### PR TITLE
Fix / tvod issues

### PR DESCRIPTION
**Pull request by royschut**
## TVOD Fixes
These are fixes to tvod, discovered during the last BeIN Sports sprint: [OTT-627]
- Inlineplayer not supporting tvod: the lock would still appear, because it didn't implement the isEntitled prop
- Subscription not reloaded after login for non-svod platforms: tvod can be implemented for any accessmodal [OTT-567]
- For premier items, the subscription offers where still shown, which shouldn't be [OTT-604]
- Did some refactoring of `useOffer()`. Wanted to push it further, but I prefer to do this separately, as we want to move more logic from the containers/hooks to the controllers/services.

## Todo
- Still need to check if drm works properly, but I don't know of a (demo) config that has this properly set up. I will do this separately, perhaps combining it with the drm-related work of @ChristiaanScheermeijer 

[OTT-627]: https://videodock.atlassian.net/browse/OTT-627?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OTT-567]: https://videodock.atlassian.net/browse/OTT-567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OTT-604]: https://videodock.atlassian.net/browse/OTT-604?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/common/src/stores/AccountController.ts [hunk 1]
```diff
@@ -509,14 +508,14 @@ export default class AccountController {
...
-  private async afterLogin(user: Customer, customerConsents: CustomerConsent[] | null, accessModel: string, shouldSubscriptionReload: boolean = true) {
+  private async afterLogin(user: Customer, customerConsents: CustomerConsent[] | null, shouldReloadSubscription = true) {
     useAccountStore.setState({
       user,
       customerConsents,
     });
 
     await Promise.allSettled([
-      accessModel === ACCESS_MODEL.SVOD && shouldSubscriptionReload ? this.reloadActiveSubscription() : Promise.resolve(),
+      shouldReloadSubscription ? this.reloadActiveSubscription() : Promise.resolve(), // For every accessModal there could be TVOD items, so we always reload the subscription
```
Comment (ChristiaanScheermeijer) [inline packages/common/src/stores/AccountController.ts]: In theory, the only call that fixes this for the TVOD offers is the `await this.refreshEntitlements?.();` in the `reloadActiveSubscription` function. 

I don't think we want to call the subscriptions endpoint when no SVOD offer id is defined. But we do want to invalidate the entitlement calls after reloading the subscription.

Comment (royschut) [reply packages/common/src/stores/AccountController.ts]: Yes, you're right. Reduced it to just the entitlements call.
I've noticed that AuthVOD + TVOD doesn't work so well yet, but this is something I'm investigating from a different ticket.

**Comment by ChristiaanScheermeijer** [comment]
I've rebased this branch

File: packages/common/src/stores/AccountController.ts [hunk 2]
```diff
@@ -509,14 +508,17 @@ export default class AccountController {
...
       user,
       customerConsents,
     });
 
+    const { accessModel } = useConfigStore.getState();
+
     await Promise.allSettled([
-      accessModel === ACCESS_MODEL.SVOD && shouldSubscriptionReload ? this.reloadActiveSubscription() : Promise.resolve(),
+      shouldReloadSubscription && accessModel === 'SVOD' ? this.reloadActiveSubscription() : Promise.resolve(),
+      shouldReloadSubscription && accessModel !== 'SVOD' ? this.refreshEntitlements?.() : Promise.resolve(), // For non-SVOD platforms, there could be TVOD items
```
Comment (ChristiaanScheermeijer) [inline packages/common/src/stores/AccountController.ts]:
```suggestion
      // For SVOD platforms, we reload the subscriptions and entitlements
      shouldReloadSubscription && accessModel === 'SVOD' ? this.reloadActiveSubscription() : this.refreshEntitlements?.(),
      // For non-SVOD platforms, there could be TVOD items, so we only the reload entitlements
      shouldReloadSubscription && accessModel !== 'SVOD' ? this.refreshEntitlements?.() : Promise.resolve(),
```

Comment (ChristiaanScheermeijer) [reply packages/common/src/stores/AccountController.ts]: Perhaps, we can rename `reloadActiveSubscription` to `reloadSubscriptions` and make let it decide if it needs to refresh SVOD subscriptions.

Then we can just call:

```suggestion
      shouldReloadSubscription ? this.reloadSubscriptions() : Promise.resolve(),
```

And check the AccessModel in the `reloadSubscriptions` function.

**Review by ChristiaanScheermeijer — APPROVED** [review]
Looks good, minor suggestions

File: packages/common/src/stores/AccountController.ts (hunk 2 above)
Comment (royschut) [reply packages/common/src/stores/AccountController.ts]: Great suggestion!

**Review by ChristiaanScheermeijer — APPROVED** [review]
🙌

**Pull request merged**